### PR TITLE
(maint) Update cask generation to support puppet-agent

### PR DESCRIPTION
**Pull request by MikaelSmith**
Updates the Rake task to generate casks to support differing versions
for puppet-agent. Add a new puppet-agent template. Regenerate all casks.

### CODE REVIEW

File: Casks/puppet-agent.rb [hunk 1]
```diff
@@ -1,26 +1,24 @@
...
-  appcast "https://downloads.puppetlabs.com/mac/puppet/#{os_ver}/x86_64/"
+
+  url "https://downloads.puppet.com/mac/puppet5/#{MacOS.version}/x86_64/puppet-agent-#{version}-1.osx#{MacOS.version}.dmg"
   name 'Puppet Agent'
   homepage "https://puppet.com/docs/puppet/#{version.major_minor}/about_agent.html"
 
-  depends_on macos: '>= :yosemite'
+  depends_on macos: '>= 10.10'
 
   pkg "puppet-agent-#{version}-installer.pkg"
```
Comment (MikaelSmith) [inline Casks/puppet-agent.rb]: pkg name needs to be updated for version switch from 5.5.4-1 to 5.5.4.

**Comment by MikaelSmith** [comment]
Skips updating pdk because we don't want to change the SHA on an existing release.

File: Casks/puppet-agent.rb [hunk 2]
```diff
@@ -1,28 +1,25 @@
...
-  
-  if MacOS.version == :yosemite
-    version '5.5.0-1'
+  case MacOS.version
+  when '10.10'
+    version '5.5.0'
     sha256 '3f30c36e9b39763839148aaea400193c7b52d8feea2765121f6dabace658ec25'
-  elsif MacOS.version == :el_capitan
-    version '5.5.0-1'
+  when '10.11'
```
Comment (lucywyman) [inline Casks/puppet-agent.rb]: @bmjen I forget if there was a reason we were using the names instead of version numbers? But I think there was a reason...

Comment (MikaelSmith) [reply Casks/puppet-agent.rb]: I should probably try this out on the different OS versions to double check they work.

Comment (MikaelSmith) [reply Casks/puppet-agent.rb]: MacOS.version behaves in interesting ways; it's default representation seems to be the version number (as a string), to get the code name as a symbol you can call `MacOS.version.to_sym`. I felt the version number, used consistently throughout the cask, was clearer.

Comment (lucywyman) [reply Casks/puppet-agent.rb]: Definitely - I initially used it as well and recall there being a reason not to. But I don't recall the reason or if it applies here.

Comment (MikaelSmith) [reply Casks/puppet-agent.rb]: Seems to work running through a few examples

### 10.10
```
$ curl http://pl-build-tools.delivery.puppetlabs.net/osx/homebrew_sierra.tar.gz | tar -x --strip 1 -C /usr/local -f -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 2059k  100 2059k    0     0  8378k      0 --:--:-- --:--:-- --:--:-- 8406k
$ brew cask install puppet-agent.rb
==> Tapping caskroom/cask
Cloning into '/usr/local/Library/Taps/caskroom/homebrew-cask'...
remote: Counting objects: 4126, done.
remote: Compressing objects: 100% (4112/4112), done.
remote: Total 4126 (delta 25), reused 822 (delta 11), pack-reused 0
Receiving objects: 100% (4126/4126), 1.30 MiB | 0 bytes/s, done.
Resolving deltas: 100% (25/25), done.
Checking connectivity... done.
Tapped 0 formulae (4,134 files, 4.1M)
==> Creating Caskroom at /usr/local/Caskroom
==> Satisfying dependencies
complete
==> Downloading https://downloads.puppet.com/mac/puppet5/10.10/x86_64/puppet-agent-5.5.0-1.osx10.10.dmg
######################################################################## 100.0%
==> Verifying checksum for Cask puppet-agent
==> Running installer for puppet-agent; your password may be necessary.
==> Package installers may write to any location; options such as --appdir are ignored.
==> installer: Package name is puppet-agent
==> installer: Installing at base path /
==> installer: The install was successful.
🍺  puppet-agent was successfully installed!
```

### 10.11
```
$ curl http://pl-build-tools.delivery.puppetlabs.net/osx/homebrew_sierra.tar.gz | tar -x --strip 1 -C /usr/local -f -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 2059k  100 2059k    0     0  2552k      0 --:--:-- --:--:-- --:--:-- 2552k
$ brew cask install puppet-bolt.rb
==> Tapping caskroom/cask
Cloning into '/usr/local/Library/Taps/caskroom/homebrew-cask'...
remote: Counting objects: 4126, done.
remote: Compressing objects: 100% (4112/4112), done.
remote: Total 4126 (delta 25), reused 822 (delta 11), pack-reused 0
Receiving objects: 100% (4126/4126), 1.30 MiB | 0 bytes/s, done.
Resolving deltas: 100% (25/25), done.
Checking connectivity... done.
Tapped 0 formulae (4,134 files, 4.0M)
==> Creating Caskroom at /usr/local/Caskroom
==> Satisfying dependencies
complete
==> Downloading https://downloads.puppet.com/mac/puppet5/10.11/x86_64/puppet-bolt-0.21.6-1.osx10.11.dmg
######################################################################## 100.0%
==> Verifying checksum for Cask puppet-bolt
==> Running installer for puppet-bolt; your password may be necessary.
==> Package installers may write to any location; options such as --appdir are ignored.
==> installer: Package name is puppet-bolt
==> installer: Installing at base path /
==> installer: The install was successful.
🍺  puppet-bolt was successfully installed!
```

### 10.12
```
$ curl http://pl-build-tools.delivery.puppetlabs.net/osx/homebrew_sierra.tar.gz | tar -x --strip 1 -C /usr/local -f -
$ brew cask install puppet-bolt.rb
==> Tapping caskroom/cask
Cloning into '/usr/local/Library/Taps/caskroom/homebrew-cask'...
remote: Counting objects: 4126, done.
remote: Compressing objects: 100% (4112/4112), done.
remote: Total 4126 (delta 25), reused 822 (delta 11), pack-reused 0
Receiving objects: 100% (4126/4126), 1.30 MiB | 0 bytes/s, done.
Resolving deltas: 100% (25/25), done.
Checking connectivity... done.
Tapped 0 formulae (4,134 files, 4.1M)
==> Creating Caskroom at /usr/local/Caskroom
==> Satisfying dependencies
complete
==> Downloading https://downloads.puppet.com/mac/puppet5/10.12/x86_64/puppet-bolt-0.21.6-1.osx10.12.dmg
######################################################################## 100.0%
==> Verifying checksum for Cask puppet-bolt
==> Running installer for puppet-bolt; your password may be necessary.
==> Package installers may write to any location; options such as --appdir are ignored.
==> installer: Package name is puppet-bolt
==> installer: Installing at base path /
==> installer: The install was successful.
🍺  puppet-bolt was successfully installed!
```

**Pull request merged**